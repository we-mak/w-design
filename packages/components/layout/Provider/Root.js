"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
Object.defineProperty(exports, "__esModule", { value: true });
var styled_1 = require("../../../utils/styled");
var theme_1 = require("../../../common/theme");
var colors = theme_1.default.colors;
exports.Root = styled_1.default.div(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n  font-family: ", ";\n\n  h1 {\n    font-size: 2rem;\n  }\n\n  h2 {\n    font-size: 1.6rem;\n  }\n\n  h3 {\n    font-size: 1.4rem;\n  }\n\n  h4 {\n    font-size: 1.2rem;\n  }\n  h5 {\n    font-size: 1rem;\n  }\n\n  h6 {\n    font-size: 0.8rem;\n  }\n\n  h1,\n  h2,\n  h3,\n  h4,\n  h5,\n  h6 {\n    font-weight: 600;\n    line-height: 1.25;\n    margin-bottom: 0.5em;\n    margin-top: 0;\n  }\n\n  h1,\n  h2,\n  h3,\n  h4 {\n    color: ", ";\n  }\n\n  h5,\n  h6 {\n    color: ", ";\n  }\n\n  p {\n    color: ", ";\n    line-height: 1.5;\n    margin-bottom: 0.5em;\n    margin-top: 0;\n  }\n\n  a {\n    color: ", ";\n    outline: none;\n    text-decoration: none;\n\n    &:hover {\n      color: ", ";\n    }\n\n    &:focus,\n    &:hover,\n    &:active {\n      text-decoration: underline;\n    }\n    &:focus,\n    &:active {\n      color: ", ";\n    }\n    &:focus {\n      text-decoration: none;\n    }\n  }\n\n  b,\n  strong {\n    font-weight: bolder;\n  }\n\n  code,\n  kbd,\n  pre,\n  samp {\n    font-family: \"Menlo, Courier, monospace\";\n  }\n\n  code {\n    color: ", ";\n    background: ", ";\n    padding: 0.1rem 0.15rem;\n    line-height: 1.2;\n  }\n\n  small {\n    font-size: 80%;\n    font-weight: 400;\n    color: ", ";\n  }\n\n  blockquote {\n    margin-left: 0;\n    padding: 0.4rem 0.8rem;\n    color: ", ";\n    p:last-child {\n      margin-bottom: 0;\n    }\n  }\n"], ["\n  font-family: ", ";\n\n  h1 {\n    font-size: 2rem;\n  }\n\n  h2 {\n    font-size: 1.6rem;\n  }\n\n  h3 {\n    font-size: 1.4rem;\n  }\n\n  h4 {\n    font-size: 1.2rem;\n  }\n  h5 {\n    font-size: 1rem;\n  }\n\n  h6 {\n    font-size: 0.8rem;\n  }\n\n  h1,\n  h2,\n  h3,\n  h4,\n  h5,\n  h6 {\n    font-weight: 600;\n    line-height: 1.25;\n    margin-bottom: 0.5em;\n    margin-top: 0;\n  }\n\n  h1,\n  h2,\n  h3,\n  h4 {\n    color: ", ";\n  }\n\n  h5,\n  h6 {\n    color: ", ";\n  }\n\n  p {\n    color: ", ";\n    line-height: 1.5;\n    margin-bottom: 0.5em;\n    margin-top: 0;\n  }\n\n  a {\n    color: ", ";\n    outline: none;\n    text-decoration: none;\n\n    &:hover {\n      color: ", ";\n    }\n\n    &:focus,\n    &:hover,\n    &:active {\n      text-decoration: underline;\n    }\n    &:focus,\n    &:active {\n      color: ", ";\n    }\n    &:focus {\n      text-decoration: none;\n    }\n  }\n\n  b,\n  strong {\n    font-weight: bolder;\n  }\n\n  code,\n  kbd,\n  pre,\n  samp {\n    font-family: \"Menlo, Courier, monospace\";\n  }\n\n  code {\n    color: ", ";\n    background: ", ";\n    padding: 0.1rem 0.15rem;\n    line-height: 1.2;\n  }\n\n  small {\n    font-size: 80%;\n    font-weight: 400;\n    color: ", ";\n  }\n\n  blockquote {\n    margin-left: 0;\n    padding: 0.4rem 0.8rem;\n    color: ", ";\n    p:last-child {\n      margin-bottom: 0;\n    }\n  }\n"])), function (props) { return props.theme.fonts; }, colors.N90, colors.N80, colors.N70, colors.B50, colors.B30, colors.B60, colors.R20, colors.N1, colors.N30, colors.N30);
exports.Root.displayName = "Root";
exports.default = exports.Root;
var templateObject_1;
//# sourceMappingURL=Root.js.map