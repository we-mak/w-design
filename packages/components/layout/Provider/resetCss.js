"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var colors_1 = require("../../../common/colors");
var ResetCss = "\n*,\n*::before,\n*::after {\n  box-sizing: inherit;\n}\n\nhtml {\n  box-sizing: border-box;\n  font-size: 20px;\n  line-height: 1.5;\n  -webkit-tap-highlight-color: transparent;\n}\n\nbody {\n  background: " + colors_1.colors.WHITE + ";\n  color: " + colors_1.colors.N60 + ";\n  font-family: -apple-system, BlinkMacSystemFont, Helvetica, \"Helvetica Neue\", system-ui, sans-serif;\n  font-size: 0.8rem;\n  overflow-x: hidden;\n  text-rendering: optimizeLegibility;\n}\n\na {\n  font-family: Helvetica, \"Helvetica Neue\", system-ui, sans-serif;\n}\n\n";
exports.default = ResetCss;
//# sourceMappingURL=resetCss.js.map