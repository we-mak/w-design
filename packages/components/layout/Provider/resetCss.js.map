{"version":3,"file":"resetCss.js","sourceRoot":"","sources":["../../../../src/components/layout/Provider/resetCss.ts"],"names":[],"mappings":";;AAAA,iDAAgD;AAEhD,IAAM,QAAQ,GAAG,oNAeD,eAAM,CAAC,KAAK,oBACjB,eAAM,CAAC,GAAG,0RAWpB,CAAC;AAEF,kBAAe,QAAQ,CAAC"}