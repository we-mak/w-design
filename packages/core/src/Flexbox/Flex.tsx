import styled from "styled-components";
import {
  space,
  justifyContent,
  flexWrap,
  flexDirection,
  alignItems,
  SpaceProps,
  JustifyContentProps,
  FlexWrapProps,
  FlexDirectionProps,
  AlignItemsProps
} from "styled-system";
import { GlobProps } from "../../../types/typings";

export interface FlexProps
  extends GlobProps,
    SpaceProps,
    JustifyContentProps,
    FlexWrapProps,
    FlexDirectionProps,
    AlignItemsProps {
  gapless?: boolean;
  oneline?: boolean;
  children?: React.ReactChild | React.ReactChild[];
}

export const Flex = styled.div<FlexProps>`
  display: flex;
  flex-wrap: wrap;
  margin-left: -0.4rem;
  margin-right: -0.4rem;
  padding: 0.4rem;
  ${space}
  ${justifyContent}
  ${flexWrap}
  ${flexDirection}
  ${alignItems}

  ${(props: FlexProps) =>
    /* set margin left and right, and child padding to 0*/
    props.gapless &&
    ` margin-left: 0 !important;
      margin-right: 0 !important;
      * {
        padding-left: 0 !important;
        padding-right: 0 !important;
      }
  `};

  ${(props: FlexProps) =>
    props.oneline &&
    `
      flex-wrap: nowrap !important;
      overflow-x: auto !important;
      -webkit-overflow-scrolling: touch;
  `};
`;

Flex.displayName = "Flex";
