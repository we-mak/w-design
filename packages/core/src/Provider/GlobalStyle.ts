import { createGlobalStyle } from "styled-components";

export const GlobalStyle = createGlobalStyle`

  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    font-weight: 700;
    line-height: 1.25;
    margin-bottom: 0.5em;
    margin-top: 0;
    font-family: ${(props: any) => props.theme.fonts.medium};
  }

  h1,
  h2,
  h3,
  h4 {
    color: ${(props: any) => props.theme.colors["N50"]};
  }

  h5,
  h6 {
    color: ${(props: any) => props.theme.colors["N60"]};
  }

  p {
    line-height: 1.5;
    margin-bottom: 0.5em;
    margin-top: 0;
    font-family: ${(props: any) => props.theme.fonts.normal};
    color: ${(props: any) => props.theme.colors["N70"]};
  }

  a {
    font-family: ${(props: any) => props.theme.fonts.normal};
    color: ${(props: any) => props.theme.colors["B50"]};
    outline: none;
    text-decoration: none;

    &:hover {
      color: ${(props: any) => props.theme.colors["B30"]};
    }

    &:focus,
    &:hover,
    &:active {
      text-decoration: underline;
    }
    &:focus,
    &:active {
      color: ${(props: any) => props.theme.colors["B60"]};
    }
    &:focus {
      text-decoration: none;
    }
  }

  small {
    font-size: 80%;
    font-weight: 400;
    color: ${(props: any) => props.theme.colors["N30"]};
  }

  blockquote {
    margin-left: 0;
    padding: 0.4rem 0.8rem;
    color: ${(props: any) => props.theme.colors["N30"]};
    p:last-child {
      margin-bottom: 0;
    }
  }

  th {
    border-bottom: 0.05rem solid ${props => props.theme.colors["N5"]};
    padding: 0.6rem 0.4rem;
    border-bottom-width: 0.1rem;
  }
  td {
    border-bottom: 0.05rem solid ${props => props.theme.colors["N2"]};
    padding: 0.6rem 0.4rem;
  }
`;
