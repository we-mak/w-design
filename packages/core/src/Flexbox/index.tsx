import * as React from "react";
import { Column } from "./Column";
import { Flex, FlexProps } from "./Flex";

class Flexbox extends React.PureComponent<FlexProps> {
  static Column = Column;

  render() {
    const { children, ...rest } = this.props;

    return (
      <Flex {...rest}>
        {React.Children.map(children, child =>
          React.cloneElement(child as React.ReactElement<HTMLElement>)
        )}
      </Flex>
    );
  }
}

export default Flexbox;
