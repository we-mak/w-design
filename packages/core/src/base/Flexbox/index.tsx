import * as React from "react";
import { Column } from "./Column";
import { Flex, FlexProps } from "./Flex";

class Flexbox extends React.PureComponent<FlexProps> {
  static Column = Column;

  render() {
    const { id, className, children, ...rest } = this.props;

    return (
      <Flex className={className} id={id} {...rest}>
        {React.Children.map(children, child =>
          React.cloneElement(child as React.ReactElement<HTMLElement>)
        )}
      </Flex>
    );
  }
}

export default Flexbox;
