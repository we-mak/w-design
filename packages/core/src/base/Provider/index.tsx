import * as React from "react";
import { ThemeProvider } from "styled-components";
import defaultTheme from "../../common/theme";
import { GlobalStyle } from "./GlobalStyle";
import "./normalize.css";

interface ThemeProps extends React.HTMLProps<HTMLDivElement> {
  theme?: object | ((theme: object) => object);
}

function Provider(p: ThemeProps) {
  const { theme = {}, children } = p;

  return (
    <ThemeProvider theme={{ ...defaultTheme, ...theme }}>
      <>
        <GlobalStyle />
        {children}
      </>
    </ThemeProvider>
  );
}

Provider.displayName = "WProvider";

export default Provider;
