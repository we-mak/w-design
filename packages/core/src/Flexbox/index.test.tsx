import * as React from "react";
import { render, cleanup } from "@testing-library/react";
import Provider from "../Provider";
import Flexbox from "./index";

afterAll(cleanup);

describe("<Flex/>", () => {
  // Render test
  it("should renders correctly", () => {
    const { container } = render(
      <Provider>
        <Flexbox>
          <Flexbox.Column />
        </Flexbox>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it("should renders gapless", () => {
    const { container } = render(
      <Provider>
        <Flexbox gapless />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it("should renders correctly", () => {
    const { container } = render(
      <Provider>
        <Flexbox oneline />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
