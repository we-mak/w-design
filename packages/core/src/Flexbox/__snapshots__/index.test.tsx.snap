// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Flex/> should renders correctly 1`] = `
<div>
  <div
    class="sc-gtssRu guoDFb"
  >
    <div
      class="sc-bdnylx bUpTWt"
    />
  </div>
</div>
`;

exports[`<Flex/> should renders correctly 2`] = `
<div>
  <div
    class="sc-gtssRu fMUBRY"
  />
</div>
`;

exports[`<Flex/> should renders gapless 1`] = `
<div>
  <div
    class="sc-gtssRu gXbIzD"
  />
</div>
`;
